handbook/engineering/web/testing.md

```diff
@@ -0,0 +1,177 @@
+# Testing web code
+
+This is the implementation of our [company testing philosophy](../testing.md) for all code (TypeScript, stylesheets, ...) used on the web, such as our webapp, browser extension, shared code, or Sourcegraph extensions.
+
+## General Goals
+
+1. **Our product and code works as intended when we ship it to customers.**
+
+2. **Our product and code doesn't accidentally break as we make changes over time.**
+
+3. **Failures should be clear about _what_ is not working and give as many hints as possible to _why_ it's not working.**
+
+4. **Tests should not commonly need manual updates when the implementation is changed.**
+   If tests commonly need manual updates, it creates the potential to accidentally change the test so that it doesn't catch a regression introduced in the same change. It also puts more burden on the code reviewers. In addition, updating tests can feel daunting and reduces the motivation to make changes to that part of the codebase.
+
+5. **Tests should be easy to write and maintain.**
+
+## Testing UI Code
+
+When testing UI, there are several categories of things that should be tested.
+These categories may be more or less important depending on the piece of UI.
+
+### Visual regressions
+
+A visual regression is a bug where the component behaves correctly, but no longer looks as intended.
+We use Percy screenshot tests to catch these.
+Percy can take screenshots in end-to-end tests, integration tests and in Storybooks.
+Storybook tests can be seen as a form of unit test for a component.
+Any story that is added to our codebase is automatically screenshotted on every CI build (but only in its initial state).
+
+### Render output
+
+Depending on the component, it can be important to test the _render output_ of the component.
+This encompasses the DOM structure and that the correct props are passed to other components.
+For example, when a component renders a link, we want to make sure right destination is passed.
+
+This is accomplished by Jest snapshot tests, using a test renderer for React like `enzyme`.
```

Comment:
I will add my thoughts a bit later on but I do consider snapshot tests for react an anti-pattern

Reply:
The idea of adding this documentation is to enable a process where you (or anyone else) can open a PR to the handbook to propose & discuss changing our practices 🙂